### ISSUE TITLE
User reported issue: change in concentration delayed

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/neuron#139]
> Problem description: Change in concentration of ions seems delayed compared to the change in membrane potential on the potential chart.
> 
> Steps to reproduce: Have Potential Chart and Concentrations showing at the same time. During depolarization, there appears to be no change in Concentrations at all. Concentrations do not start changing until after 3 milliseconds.
> 
> Severity: Seemed to be uniform issue across all student computers. I feel as if the previous version of the neuron simulation was more sensitive in change in concentration or at least better timed to reflect the action potential chart.

@phet-steele and I discussed the possibility of adding decimal places, but it is not clear that would resolve the issue.

**Comment by phet-steele** [comment]
> @phet-steele and I discussed the possibility of adding decimal places, but it is not clear that would resolve the issue.

It wouldn't! I tried 20 decimal places and it didn't help this.

<img width="1039" alt="screen shot 2017-10-10 at 3 36 28 pm" src="https://user-images.githubusercontent.com/5863899/31412037-d13033ea-add0-11e7-9388-fdd50a28a72d.png">

**Comment by jbphet** [comment]
One of the things the user said was:

> I feel as if the previous version of the neuron simulation was more sensitive in change in concentration or at least better timed to reflect the action potential chart.

I just did a comparison between the Java and HTML5 sims, and the results are captured in the screenshots shown below.  The procedure was to stimulate the neuron, pause as quickly as I could after seeing a change in concentration, then use the forward and backward step buttons to move to the time step when the first change of concentration occurred for any of the readouts.  The Na+ value outside the membrane is the first thing that changes, and it goes from 150.00000 to 149.99999 mM.  It appears that the changes are very close to being in the same place in both sims.  In my opinion, the differences visible here are not significant enough that it would impact the pedagogy.

Screenshots:

![image](https://user-images.githubusercontent.com/3082831/54396172-ba2bdc80-4677-11e9-8200-a97ca2c6c0a2.png)

![image](https://user-images.githubusercontent.com/3082831/54396203-d2036080-4677-11e9-86f7-ca207254d21d.png)
![image](https://user-images.githubusercontent.com/3082831/54396701-7f2aa880-4679-11e9-885f-287fa5fb9953.png)

**Comment by jbphet** [comment]
As an experiment, I tried 50 digits of resolution, and that doesn't have a very significant effect.  Here is what that looks like (same procedure as in the previous comment):

![image](https://user-images.githubusercontent.com/3082831/54397071-cbc2b380-467a-11e9-9328-c6b2578ff242.png)

**Comment by jbphet** [comment]
I took a few minutes and refamiliarized myself with the model code, and there is an intentional parameter in the code that introduces some delay into the updating of the concentration readouts so that the user doesn't see these readout values change until some of the animated sodium and potassium ions have had time to cross the membrane.  As an experiment, I reduced the value of this code parameter, and indeed, the concentration values changed sooner.  I'm just not sure if this is a good thing to do in the published version.

We had a lot of help from outside of PhET when this sim was originally designed, but @amanda-phet helped with the redesign, and I'm not sure who else to ask.  So, hey @amanda-phet - what are your thoughts?  Would it be better to make the concentrations change sooner, which would be more technically accurate, but could potentially be a bit confusing since the animated particles might not have crossed the membrane when the readouts change?

Personally, I think I would opt for what the user suggests, and make the changes to concentration happen sooner.  My thinking is that it is more accurate to a real action potential, and I think the students who use this sim - who will likely be high school students or later - will understand that the particles are animations and they aren't seeing all of them.

**Comment by jbphet** [comment]
Note to self: If we do decide to change this, the constant is `CONCENTRATION_READOUT_DELAY` in `NeuronModel.js`.

**Comment by amanda-phet** [comment]
I am going to see if Mike Klymkowsky is available to discuss this. I personally don't see an issue with the current version, but I am far from an appropriate pedagogical expert on this topic, and if a teacher is raising the issue it seems worth investigating.

**Comment by jbphet** [comment]
@amanda-phet and I discussed this with Mike K, and he didn't have a strong opinion either way.  @amanda-phet and I then worked through an action potential slowly, and determined that it would be reasonable to reduce the delay for the concentration changes to half of its current value, since particles are clearly crossing the membrane by then.  We tried reducing this delay all the way to zero, and that looked wrong because the concentration values changed before particles had crossed, and in fact before the gates were fully opened.
